pp_hot.c

```diff
@@ -123,6 +123,53 @@ PP(pp_and)
     }
 }
 
+/*
+ * Mashup of simple padsv + sassign OPs
+ * Doesn't support the following lengthy and unlikely sassign case:
+ *    (UNLIKELY(PL_op->op_private & OPpASSIGN_CV_TO_GV))
+ *  These cases have a separate optimization, so are not handled here:
+ *    (PL_op->op_private & OPpASSIGN_BACKWARDS) {or,and,dor}assign
+*/
+
+PP(pp_padsv_store)
+{
+    dSP;
+    OP * const op = PL_op;
+    SV** const padentry = &PAD_SVl(op->op_targ);
+    SV* targ = *padentry; /* lvalue to assign into */
+    SV* const val = TOPs; /* RHS value to assign */
+
+    /* !OPf_STACKED is handled by pp_padsv2padsv */
+    assert(op->op_flags & OPf_STACKED);
+
+    /* Inlined, simplified pp_padsv here */
+    if (op->op_private & OPpLVAL_INTRO) {
+        if (!(op->op_private & OPpPAD_STATE)) {
+            save_clearsv(padentry);
+        }
+    }
+    if (op->op_private & OPpDEREF) {
+        targ = vivify_ref(targ, op->op_private & OPpDEREF);
+    }
+
+    /* Inlined, simplified pp_assign from here */
```

Comment:
pp_sassign

Reply:
Thanks, updated in my local copy until I fixup the ASAN errors and rebase.